features/step_definitions/location_steps.rb

```diff
@@ -17,7 +17,7 @@
   location.name.click
 end
 
-When(/^I view the details of an appointment location that handles its own booking$/) do
+When(/^(?:I view|a user views) the details of an appointment location(?: that handles its own booking)?$/) do
```

Comment:
Perhaps define a separate `When` instead of introducing complexity into the regex here?